hazelcast/src/main/java/com/hazelcast/wan/impl/merkletree/ArrayMerkleTree.java

```diff
@@ -81,32 +80,29 @@
  * {@link MerkleTreeUtil#sumHash(int, int)}
  */
 public class ArrayMerkleTree extends AbstractMerkleTreeView implements MerkleTree {
-    private final OAHashSet<Object>[] leafKeys;
     private final int leafLevel;
 
     /**
-     * Footprint holds the total memory footprint of the Merkle tree and
-     * the leaf data blocks.
-     * <p>
-     * Note that this field leverages a single-writer and a non-atomic
-     * operation is executed on the field. See {@link #adjustFootprintWithLeafKeySetChange}
+     * Footprint holds the total memory footprint of the Merkle tree.
      */
-    private volatile long footprint;
+    private final long footprint;
 
-    @SuppressWarnings("unchecked")
     public ArrayMerkleTree(int depth) {
         super(depth);
-
         this.leafLevel = depth - 1;
-
-        final int leaves = MerkleTreeUtil.getNodesOnLevel(leafLevel);
-
-        leafKeys = new OAHashSet[leaves];
-        for (int i = 0; i < leaves; i++) {
-            leafKeys[i] = new OAHashSet<Object>(1);
-        }
-
-        initializeFootprint();
+        this.footprint = INT_SIZE_IN_BYTES * tree.length
+                // reference to the tree
+                + REFERENCE_COST_IN_BYTES
+                // reference to leafKeys array
```

Comment:
I think we don't have this anymore since `leafKeys` is removed

Reply:
Removed.